### PR TITLE
Convert privacy plugins to service provider

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Converts the media action plugins to service providers.

### Testing Instructions
Create some contacts and assign a user to them, then make an export request and export the contact data. Same for content.

### Actual result BEFORE applying this Pull Request
All works.

### Expected result AFTER applying this Pull Request
All works.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [x] Pull Request link for manual.joomla.org: https://github.com/joomla/Manual/pull/94
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: plugins/privacy/contact/services/provider.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Extension\PluginInterface;
+use Joomla\CMS\Factory;
+use Joomla\CMS\Plugin\PluginHelper;
+use Joomla\Database\DatabaseInterface;
+use Joomla\DI\Container;
+use Joomla\DI\ServiceProviderInterface;
+use Joomla\Event\DispatcherInterface;
+use Joomla\Plugin\Content\Contact\Extension\Contact;
```
Comment (heelc29) [inline plugins/privacy/contact/services/provider.php]:
```suggestion
use Joomla\Plugin\Privacy\Contact\Extension\Contact;
```

File: plugins/privacy/content/services/provider.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Extension\PluginInterface;
+use Joomla\CMS\Factory;
+use Joomla\CMS\Plugin\PluginHelper;
+use Joomla\Database\DatabaseInterface;
+use Joomla\DI\Container;
+use Joomla\DI\ServiceProviderInterface;
+use Joomla\Event\DispatcherInterface;
+use Joomla\Plugin\Content\Contact\Extension\Contact;
```
Comment (heelc29) [inline plugins/privacy/content/services/provider.php]:
```suggestion
use Joomla\Plugin\Privacy\Content\Extension\Content;
```

File: plugins/privacy/content/services/provider.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function register(Container $container): void
+    {
+        $container->set(
+            PluginInterface::class,
+            function (Container $container) {
+                $dispatcher = $container->get(DispatcherInterface::class);
+                $plugin     = new Contact(
```
Comment (heelc29) [inline plugins/privacy/content/services/provider.php]:
```suggestion
                $plugin     = new Content(
```

File: plugins/privacy/message/services/provider.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Extension\PluginInterface;
+use Joomla\CMS\Factory;
+use Joomla\CMS\Plugin\PluginHelper;
+use Joomla\Database\DatabaseInterface;
+use Joomla\DI\Container;
+use Joomla\DI\ServiceProviderInterface;
+use Joomla\Event\DispatcherInterface;
+use Joomla\Plugin\Privacy\Content\Extension\Message;
```
Comment (heelc29) [inline plugins/privacy/message/services/provider.php]:
```suggestion
use Joomla\Plugin\Privacy\Message\Extension\Message;
```

File: plugins/privacy/message/src/Extension/Message.php [hunk 5]
```diff
@@ -6,10 +6,10 @@
  *
  * @copyright   (C) 2018 Open Source Matters, Inc. <https://www.joomla.org>
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
-
- * @phpcs:disable PSR1.Classes.ClassDeclaration.MissingNamespace
  */
 
+namespace Joomla\Plugin\Privacy\Content\Extension;
```
Comment (heelc29) [inline plugins/privacy/message/src/Extension/Message.php]:
```suggestion
namespace Joomla\Plugin\Privacy\Message\Extension;
```

**Review by heelc29 — CHANGES_REQUESTED** [review]

File: administrator/components/com_privacy/src/Plugin/PrivacyPlugin.php [hunk 6]
```diff
@@ -28,11 +29,14 @@
...
 abstract class PrivacyPlugin extends CMSPlugin
 {
+    use DatabaseAwareTrait;
+
     /**
      * Database object
      *
      * @var    \Joomla\Database\DatabaseDriver
      * @since  3.9.0
+     * @deprecated  __DEPLOY_VERSION__ will be removed in 6.0 use $this->getDatabase() instead
```
Comment (heelc29) [inline administrator/components/com_privacy/src/Plugin/PrivacyPlugin.php:39]: documentation changes for manual.joomla.org needed?

**Comment by heelc29** [comment]
I have tested this item :red_circle: unsuccessfully on 8110a9d2856a04a697b0bf44ea5742d58b4e14db<br /><br />During export data `Class "Joomla\Plugin\Privacy\User\Extension\UserPlugin" not found`<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40174">issues.joomla.org/tracker/joomla-cms/40174</a>.</sub>

File: plugins/privacy/user/src/Extension/User.php [hunk 7]
```diff
@@ -6,11 +6,10 @@
...
-
- * @phpcs:disable PSR1.Classes.ClassDeclaration.MissingNamespace
  */
 
-use Joomla\CMS\Application\CMSApplicationInterface;
+namespace Joomla\Plugin\Privacy\User\Extension;
+
 use Joomla\CMS\Language\Text;
 use Joomla\CMS\Table\User as TableUser;
 use Joomla\CMS\User\User;
```
Comment (heelc29) [inline plugins/privacy/user/src/Extension/User.php]: Is it possible to use `Joomla\CMS\User\User` with an alias, so the class of the plugin can be `User` (`Joomla\Plugin\Privacy\User\Extension\User`)?

Comment (laoneo) [reply plugins/privacy/user/src/Extension/User.php]: It was easier to do it that way 😁

Comment (heelc29) [reply plugins/privacy/user/src/Extension/User.php]: But then the file must be renamed ... `User.php` to `UserPlugin.php`

Comment (laoneo) [reply plugins/privacy/user/src/Extension/User.php]: Yes sure, should not do pr's when I'm on traveling :-)

Comment (laoneo) [reply plugins/privacy/user/src/Extension/User.php]: renamed

**Comment by heelc29** [comment]
I have tested this item :white_check_mark: successfully on 30f40286832433a94bc35d5720faf9e98320040c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40174">issues.joomla.org/tracker/joomla-cms/40174</a>.</sub>

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on 30f40286832433a94bc35d5720faf9e98320040c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40174">issues.joomla.org/tracker/joomla-cms/40174</a>.</sub>

**Comment by joomdonation** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40174">issues.joomla.org/tracker/joomla-cms/40174</a>.</sub>

File: administrator/components/com_privacy/src/Plugin/PrivacyPlugin.php (hunk 6 above)
Comment (laoneo) [reply administrator/components/com_privacy/src/Plugin/PrivacyPlugin.php:39]: True, check https://github.com/joomla/Manual/pull/94.

**Pull request merged**